lib/hijri_calendar.dart

```diff
@@ -218,49 +224,49 @@ class HijriCalendar {
       yearString = year.toString();
     }
 
-    if (newFormat.contains("dd")) {
+    if (format.contains("dd")) {
       newFormat = newFormat.replaceFirst("dd", dayString);
     } else {
-      if (newFormat.contains("d")) {
+      if (format.contains("d")) {
         newFormat = newFormat.replaceFirst("d", day.toString());
       }
     }
 
     //=========== Day Name =============//
     // Friday
-    if (newFormat.contains("DDDD")) {
+    if (format.contains("DDDD")) {
       newFormat = newFormat.replaceFirst(
           "DDDD", "${_local[language]!['days']![wkDay ?? weekDay()]}");
 
       // Fri
-    } else if (newFormat.contains("DD")) {
+    } else if (format.contains("DD")) {
       newFormat = newFormat.replaceFirst(
           "DD", "${_local[language]!['short_days']![wkDay ?? weekDay()]}");
     }
 
     //============== Month ========================//
     // 1
-    if (newFormat.contains("mm")) {
+    if (format.contains("mm")) {
       newFormat = newFormat.replaceFirst("mm", monthString);
-    } else {
+    } else if (format.contains("m")){
```

Comment:
So do you suggest writing a final `else` to cover unpredicted use cases? Or do you mean wring a test code that checks possible errors in the package?

Reply:
It would be great if you cover the else preventing any unpredicted case,
Regarding the tests, there are already many tests are written in the package already, you may take it as an example and write similar tests.

Reply:
I didn't apply these changes because they are not relevant with Turkish language support. I'll happily contribute with another PR for these fixes.